UM/Settings/Interfaces.py

```diff
@@ -81,25 +85,39 @@ def hasProperty(self, key: str, property_name: str) -> bool:
     #   The serialized representation of the container can be used to write the
     #   container to disk or send it over the network.
     #
-    #   \param ignored_metadata_keys A list of keys that should be ignored when it serializes the metadata.
+    #   \param ignored_metadata_keys A set of keys that should be ignored when
+    #   it serializes the metadata.
     #
     #   \return \type{string} A string representation of this container.
-    def serialize(self, ignored_metadata_keys: Optional[List] = None) -> str:
+    def serialize(self, ignored_metadata_keys: Optional[set] = None) -> str:
```

Comment:
The typing is not yet consistent in all places:

```
./UM/Settings/SettingDefinition.py:    def serialize(self) -> str:
./UM/Settings/Interfaces.py:    def serialize(self, ignored_metadata_keys: Optional[set] = None) -> str:
./UM/Settings/ContainerRegistry.py:    def serialize(self, ignored_metadata_keys: Optional[set] = None) -> str:
./UM/Settings/InstanceContainer.py:    def serialize(self, ignored_metadata_keys: Optional[set] = None) -> str:
./UM/Settings/DefinitionContainer.py:    def serialize(self, ignored_metadata_keys: Optional[set] = None):
./UM/Settings/ContainerStack.py:    def serialize(self, ignored_metadata_keys: Optional[set] = None):
./tests/Settings/ContainerTestPlugin/ContainerTestPlugin.py:    def serialize(self, ignored_metadata_keys=set()):
./tests/Settings/TestContainerRegistry.py:    def serialize(self, ignored_metadata_keys = None):
./tests/Settings/TestContainerStack.py:    def serialize(self, ignored_metadata_keys = None)
```

Reply:
I'll check them all:

* `SettingDefinition` is not in this class' family tree.
* `ContainerRegistry._EmptyInstanceContainer` is consistent.
* `InstanceContainer` is consistent.
* `DefinitionContainer` is consistent.
* `ContainerStack` is consistent.
* `ContainerTestPlugin` is consistent.
* `TestContainerRegistry` is consistent.
* `TestContainerStack` is consistent.

So it seems that no change is needed (or not any more).